### PR TITLE
Add Android update screen

### PR DESCRIPTION
**Pull request by lubej**
## Solution

Adds an update screen, when update is available on mobile platforms(currently only implemented for Android platform). Where clicking on "Update now" takes the user to the app store page.

## Resources

localhost_3000_(iPhone 12 Pro)

### CODE REVIEW

**Comment by lubej** [comment]
Note to self: Does update check run after user enters the wallet again. It should check only once, and then maybe once per day?

**Comment by lubej** [comment]
> Note to self: Does update check run after user enters the wallet again. It should check only once, and then maybe once per day?

Current implementation checks for an update only on app launch. Which should be sufficient for current case, as the update can be skipped anyway. When mandatory update feature start to roll out, we should revisit the topic on what time interval the update check should be implemented.

File: src/app/components/Ionic/providers/IonicProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+const ionicProviderInitialState: IonicProviderState = {
+  updateAvailability: UpdateAvailability.NOT_INITIALIZED,
+  error: null,
+}
+
+export const IonicContextProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [state, setState] = useState<IonicProviderState>({ ...ionicProviderInitialState })
+
+  const { checkForUpdateAvailability } = useIonicRequiresUpdate(state, setState)
```
Comment (lukaw3d) [inline src/app/components/Ionic/providers/IonicProvider.tsx:15]: Sending changes into setState like this is hard to trace

**Review by lukaw3d — APPROVED** [review]

File: src/app/components/Ionic/providers/IonicProvider.tsx (hunk 1 above)
Comment (lubej) [reply src/app/components/Ionic/providers/IonicProvider.tsx:15]: Yes I agree. But making a redux slice just for this seemed like an overkill.

Comment (lukaw3d) [reply src/app/components/Ionic/providers/IonicProvider.tsx]: ew i didn't mean that :D just move useState into useIonicRequiresUpdate and return state out

Comment (lubej) [reply src/app/components/Ionic/providers/IonicProvider.tsx]: Right, my bad 😄 

Here is the follow up for it: https://github.com/oasisprotocol/oasis-wallet-web/pull/1969

**Pull request merged**